alacritty_terminal/src/grid/mod.rs

```diff
@@ -62,7 +60,7 @@ impl<T: PartialEq> ::std::cmp::PartialEq for Grid<T> {
             && self.cols.eq(&other.cols)
             && self.lines.eq(&other.lines)
             && self.display_offset.eq(&other.display_offset)
-            && self.scroll_limit.eq(&other.scroll_limit)
+            && self.history_size().eq(&other.history_size())
```

Comment:
Do we need this comparison? It should be automatically assured when the raw buffer is compared, right? So I'm not sure if this can ever be `false`.

Reply:
It could be false, if your len on a storage defer, since we're matching only by inner.len() etc.

Reply:
Shouldn't we check the `len` in storage then?

Reply:
Shouldn't we check the `len` in `storage.rs` then?

Reply:
I think we can.

Reply:
We should do this, querying `self.history_size()` in this function does not make sense.